src/rotary-encoder.cpp

```diff
@@ -43,9 +49,50 @@ ISR(INT1_vect)
     if (t - last_trigger_INT1 > 70)
     {
         if (PIND & bit(DT_PIN))
-            ccw_rotation();
+            queuing_interrupt(CCW_INTERRUPT);
         else
-            cw_rotation();
+            queuing_interrupt(CW_INTERRUPT);;
     }
     last_trigger_INT1 = t;
 }
+
+void queuing_interrupt(int interrupt)
+{
+    // Create a new interrupt
+    struct Interrupt newInterrupt;
+    newInterrupt.type = interrupt;
+
+    // Check if the queue is full
+    if ((rear + 1) % QUEUE_SIZE != front) {
+        // Add the interrupt to the queue
+        interruptQueue[rear] = newInterrupt;
+        rear = (rear + 1) % QUEUE_SIZE;
+    } else {
+        // Handle queue overflow (optional)
+    }
+}
+
+
+
+void dequeuing_interrupt()
+{
+    // Check if there are pending interrupts in the queue
+    if (front != rear) {
```

Comment:
If you invert the logic of the check, you can exit early and reduce the indentation of the function, which makes it easier to read (indentation in general increases the perceived complexity):

```c
    if (front == rear) {
        return;
    }
```

If you, at the same time make it a named function, then you can also remove the comment, which makes the code's intent easier to read.

```suggestion
    if (event_queue_is_empty(&queue)) {
        return;
    }
```

Reply:
As you can imagine, I did a lot of this with ChatGPT 😏.
When working with the front and rear variables I noticed that both increase and none of them decrease. I wonder if they would overflow at some point.?